samples/testing/testify/suite_test.go

```diff
@@ -0,0 +1,63 @@
+package testify
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+)
+
+// Interface to be tested
+type Doubler interface {
+	Double(int) int
+}
+
+// Implementation 1 (using addition)
+type ByAdd struct{}
+
+func (b ByAdd) Double(x int) int {
+	return x + x
+}
+
+// Implementation 2 (using multiplication)
+type ByMul struct{}
+
+func (b ByMul) Double(x int) int {
+	return x * 2
+}
+
+// Suite definition
+type DoublerSuite struct {
+	suite.Suite
+	Doubler
+}
+
+func TestDoubler(t *testing.T) {
+	// Create the suites with different implementations
+	byAddSuite := &DoublerSuite{
+		Doubler: ByAdd{},
+	}
+	byMulSuite := &DoublerSuite{
+		Doubler: ByMul{},
+	}
```

Comment:
I'd put those on a single line (not that much going on):
```go
	byAddSuite := &DoublerSuite{Doubler: ByAdd{}}
	byMulSuite := &DoublerSuite{Doubler: ByMul{}}
```

Reply:
Done